### PR TITLE
Cleanup apiserversource docs

### LINKED ISSUE #3351: Clean up APIServerSource docs
Currently, there seem to be two different docs for APIServerSource:
- https://knative.dev/development/eventing/sources/apiserversource/
- https://knative.dev/development/eventing/samples/kubernetes-event-source/

Tasks required:

- [x] We should  combine these into one well written doc with good examples under the event sources section.
- [x] Add alias as required to reroute from old sample doc, check for and update any broken links due to the change.
- [x] Add the kn CLI commands.

### PR DESCRIPTION
**Pull request by abrennan89**
Fixes #3351

### CODE REVIEW

**Comment by abrennan89** [comment]
/hold

Awaiting reviews

File: docs/eventing/sources/apiserversource/getting-started/_index.md
Comment (n3wscott) [inline docs/eventing/sources/apiserversource/getting-started/_index.md:56]: For yaml, I think we should standardize on this style:

```shell
kubectl create -f - <<EOF
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
    name: <cluster_role_binding_name>
  roleRef:
    apiGroup: rbac.authorization.k8s.io
    kind: ClusterRole
    name: <your_cluster_role>
  subjects:
  - kind: ServiceAccount
    name: <your_service_account>
    namespace: <your_namespace>
EOF
```

What do you think?

Then the tabs are `kubectl` or `kn`

File: docs/eventing/sources/apiserversource/getting-started/_index.md
Comment (n3wscott) [inline docs/eventing/sources/apiserversource/getting-started/_index.md:32]: I am not a big fan of the `<service_account_name>` usage, it means the person following the demo can't copy and paste without edits. 

I would recommend us either:

1. Define a block of exports that will be used in the demo, like:

```shell
export service_account_name=example-serviceaccountname
```

then used 

```shell
      name: ${service_account_name}
```

Or 2. 

we just make up a demoish name that does not conflict with our other demos.

```yaml
      name: apiserversource-demo
```

Comment (abrennan89) [reply docs/eventing/sources/apiserversource/getting-started/_index.md:32]: I guess it's just difference of opinion, but I feel like people should be using their own values and docs as a reference rather than just copy and pasting a whole dummy environment. 

@omerbensaadon do you have any UX research about this either way? Maybe people do just want to copy/paste commands but to me that just doesn't add any value as opposed to explaining what things actually are (i.e. teaching them to fish). Also, having callouts maybe would solve this problem because we could use the demo values but then still give an explanation... sort of why I was asking about that in Slack FYI 🙂

File: docs/eventing/sources/apiserversource/getting-started/_index.md
Comment (omerbensaadon) [inline docs/eventing/sources/apiserversource/getting-started/_index.md:12]: Note to self: we may want to add a link to Prerequisites page here once the split (#3399) is merged

File: docs/eventing/sources/apiserversource/getting-started/_index.md
Comment (omerbensaadon) [reply docs/eventing/sources/apiserversource/getting-started/_index.md:32]: @abrennan89 having callouts would totally solve this!

I don't have feedback specific to this from user interviews, but maybe there's a happy medium between these two ideas? 

Here's what Samia + Richard and I did on the last project we worked on: 
Screen Shot 2021-03-31 at 9 12 23 AM


1. The code block 
2. Where: and then a short description of each field? 

Maybe the way Scott has structured it (`${service_account_name}`) would make it obvious to folks that they've got to throw their own values in? 

@snneji and @RichardJJG may have opinions here though, I'm confident they've dealt with this sort of thing before...

Comment (omerbensaadon) [reply docs/eventing/sources/apiserversource/getting-started/_index.md:32]: @abrennan89 let me know if you like this idea, I am happy to take a crack at implementing :)

File: docs/eventing/sources/apiserversource/getting-started/_index.md
Comment (omerbensaadon) [reply docs/eventing/sources/apiserversource/getting-started/_index.md:56]: If the rest of the YAML on the site looks like what you posted Ashleigh, I would keep it. 

I do, however, agree that we should tell people how/when to run the files we have them create?

File: docs/eventing/sources/apiserversource/getting-started/_index.md
Comment (abrennan89) [reply docs/eventing/sources/apiserversource/getting-started/_index.md:12]: I think we can do that in a follow up PR at some point, maybe create an issue to standardize all the prerequisites for procedures?

**Comment by abrennan89** [comment]
@omerbensaadon a nice issue we could do maybe to standardize this going forward would be to propose:

- Standardizing ways to lay out YAML, i.e. decide if we want to add the kubectl commands for all, and if so, implement it consistently across the site.
- Standardize a way to format variables / user added values in examples.
- Standardize callouts / descriptions as you have suggested above. I think what you're suggesting is a reasonable stop gap for now (pretty much how I've dealt with it in the past too) and then create an issue as part of the mkdocs move to add callout formatting properly later?

File: docs/eventing/sources/apiserversource/getting-started/_index.md
Comment (omerbensaadon) [reply docs/eventing/sources/apiserversource/getting-started/_index.md:12]: Abso-lutely :) I'll set up an issue after #3399 gets merged

**Comment by omerbensaadon** [comment]
@abrennan89 +1 to all of that, where should we capture this new standard?

**Comment by n3wscott** [comment]
We should iterate a few times and nail down what we want for the rest of eventing. Every change is better than we had before. So each update is getting us closer to 🎉

**Comment by omerbensaadon** [comment]
@n3wscott totally agree, this shouldn't be a blocker but it's something we should begin to capture as we commit to a Style Guide for docs 👍🏼 

I think this is something that Richard and Samia also struggled with upon joining, like what are things supposed to look like?

**Comment by abrennan89** [comment]
@omerbensaadon let's maybe start working on the style stuff tomorrow afternoon if I can get some time; I'll try to pull something together before we chat and we can make any changes to this PR that are practical to do in the meantime without blocking it for too long. I'll maybe add our "makeshift callouts" i.e. descriptions like you have above for now, and we should look at things like how the k8s style guide does variables/values that can be modified in code as a jumping off point.

/hold

File: docs/eventing/sources/apiserversource/getting-started/_index.md
Comment (abrennan89) [reply docs/eventing/sources/apiserversource/getting-started/_index.md:56]: It might be worth rather than making the whole thing copy/paste, giving the `kubectl apply` commands and the YAML as proper steps and add the variables and descriptions for the YAML/commands. We can talk about it in person tomorrow.

File: docs/eventing/sources/apiserversource/getting-started/_index.md
Comment (snneji) [reply docs/eventing/sources/apiserversource/getting-started/_index.md:32]: Unless I'm mistaken, I think this is already in Google's style guide. Google's recommendation says placeholder variables should be in the format `SERVICE_ACCOUNT_NAME`. See [Formatting placeholders](https://developers.google.com/style/placeholders). Callouts are also mentioned at the end of that topic.

File: docs/eventing/sources/apiserversource/getting-started/_index.md
Comment (snneji) [inline docs/eventing/sources/apiserversource/getting-started/_index.md:64]: Also the Google style guide topic I linked above discourages using possessive pronouns in placeholder variables. So instead of `<your_cluster_role>` Google recommends `CLUSTER_ROLE_NAME`.

File: docs/eventing/sources/apiserversource/getting-started/_index.md
Comment (abrennan89) [reply docs/eventing/sources/apiserversource/getting-started/_index.md:56]: lol @omerbensaadon we didn't talk about any standardization stuff on our call like I thought we would; do we want to create a proposal re the style guide stuff too? Can't remember if we decided on that after the long weekend.

File: docs/eventing/sources/apiserversource/getting-started/_index.md
Comment (abrennan89) [reply docs/eventing/sources/apiserversource/getting-started/_index.md:32]: @omerbensaadon this is different to how Kubernetes does variables - I prefer the k8s way over the Google style guide, but I think these sorts of clashes are why we should have a Knative specific style guide that cements which ones we're using between k8s and Google styles.
https://kubernetes.io/docs/contribute/style/style-guide/#use-angle-brackets-for-placeholders

File: docs/eventing/sources/apiserversource/getting-started/_index.md
Comment (abrennan89) [reply docs/eventing/sources/apiserversource/getting-started/_index.md:12]: @omerbensaadon ping - reminder to create the issue since it looks like that's merged now ;)

File: docs/eventing/sources/apiserversource/getting-started/_index.md
Comment (abrennan89) [reply docs/eventing/sources/apiserversource/getting-started/_index.md:64]: that seems acceptable, I just wasn't sure what to call it

Comment (abrennan89) [reply docs/eventing/sources/apiserversource/getting-started/_index.md:64]: I'm going to go with <this-format> as per https://kubernetes.io/docs/contribute/style/style-guide/#use-angle-brackets-for-placeholders but I will update the names 🙂

**Review by n3wscott — APPROVED** [review]
/lgtm
/approve

**Comment by omerbensaadon** [comment]
@abrennan89 just getting the time to look at this and it looks fantastic. 

We should link to this as one of the potential "Next Steps" after the Getting Started Guide. 

I am going to capture some of the style choices you've made here in [Knative Style Guide Braindump](https://docs.google.com/document/u/1/d/1tigolRzKhronvgoAJfszE-XHWWd0LQ8o_j0D73O9KvM/edit#)

**Pull request merged**
